cmd/frontend/db/event_logs.go

```diff
@@ -312,6 +360,52 @@ func (l *eventLogs) countPerPeriodBySQL(ctx context.Context, countExpr, interval
 	return counts, nil
 }
 
+func (l *eventLogs) calculatePercentilesPerPeriodBySQL(ctx context.Context, interval, period *sqlf.Query, startDate, endDate time.Time, field string, percentiles []float64, conds []*sqlf.Query) ([]PercentileValue, error) {
+	countByPeriodExprs := []*sqlf.Query{}
+	qExprs := []*sqlf.Query{}
+	for i, p := range percentiles {
+		name := fmt.Sprintf("p%d\n", i)
+		countByPeriodExprs = append(countByPeriodExprs, sqlf.Sprintf("percentile_cont(%d) WITHIN GROUP (ORDER BY (argument->'"+field+"')::integer) AS "+name, p))
```

Comment:
Do we have a GIN index on the arguments column? Seems appropriate.

Reply:
👍 

We already have a lot of indices on this table, and it could have a lot of rows as discussed in slack (for dotcom and for large customers). I see the need here of course, but given the use cases for these queries (infrequent/period reports for pings, and dashboards that only site-admins will have access to) perhaps that tradeoff is less clear than for core functionality queries.

Reply:
Could `field` be a source of SQL injection? 

Can we do this instead? (I don't know if Postgres complaints):
```suggestion
		countByPeriodExprs = append(countByPeriodExprs, sqlf.Sprintf("percentile_cont(%d) WITHIN GROUP (ORDER BY (argument->'%s')::integer) AS "+name, field, p))
```

If not possible doing above, how about we have a whitelisted candidates of the argument `field`?

Reply:
Field will be supplied via the usagestats package and should only be compile-time constants. It seems like I can't get sqlf to quote this appropriately as it's not something that can be supplied as a parameter (at least using this syntax).

I could be proactive about this but I'm not sure what the best solution is (whitelist, comments, etc).

Reply:
Nvm, updated. This does work (but replacing _name_ with a parameter does not).